src/cost_functions.jl

```diff
@@ -54,17 +58,33 @@ function (f::L2Loss)(sol::DESolution)
       for j in 1:length(sol[i])
         sumsq +=(data[j,i] - sol[j,i])^2
       end
+      if diff_data != nothing
```

Comment:
since we don't need this data, we can just do this off of whether a weight is given.

Reply:
But it then requires user to always give weights incase they want to use the first difference too, which probably will not be the case mostly?

Reply:
I mean between giving weights and not giving weights, in my opinion not having weights would be the more common one?

Reply:
Yeah, I'm not sure. Maybe just set the weight to 1 by default and just always do the calculation unless it's zero.